### DISCUSSION TITLE
Router modification

### DISCUSSION THREAD

**Discussion by bellaabdelouahab** [ISIL-ESTE/Student-Workflow-Organizer#50] [Q&A] [answered]
is this still good :
```js
const authRoutes = (mainrouter) => {
    router.post("/signup", authController.signup);
    router.post("/login", authController.login);
    mainrouter.use("/auth", router);
}
module.exports = authRoutes;
```
because I m trying to make the swagger and also fix code structure and i found that this is the best way

**Comment by bellaabdelouahab** [comment]
@Khalid1G @muttaqin1

**Comment by ghost** [comment]
Yeah, looks perfect 💯
@bellaabdelouahab

**Reply by bellaabdelouahab** [reply]
Well after 6 hours of coding and searching, i can assure you that swager will never be fully generic like spring boot 😂😂😂

However I managed to make it generic in terms of endpoints but not in properties, you still need to configur the body
@muttaqin1

**Reply by ghost** [reply]
I have also done some research and came up with the same conclusion😁
@bellaabdelouahab

**Reply by ghost** [reply, marked as answer]
got it thanks😁 @bellaabdelouahab
